scripts/validate_string_concatenation.py

```diff
@@ -102,28 +106,157 @@ def strings_to_concatenate(source_path: str) -> Generator[Tuple[str, int], None,
         Source file path.
     line_number : int
         Line number of unconcatenated string.
+    MSG : str
+        Explenation of the error.
+
+    Notes
+    -----
+    GH #30454
     """
+    MSG: str = (
+        "String unnecessarily split in two by black. Please merge them manually."
+    )
```

Comment:
I'd personally have the string directly where it's used, but up to you, it's just a preference.